### PR TITLE
First cut keyboard support

### PR DESCRIPTION
**Pull request by roscopeco**
WIP as **very** lightly tested currently, and probably needs more work in libs / software.

### CODE REVIEW

**Review by 0xTJ — COMMENTED** [review]
Overall it looks good! I've just got one comment on documentation and a less important note on the way that keyboard detection is written.

File: code/firmware/rosco_m68k_firmware/trap14.asm [hunk 1]
```diff
@@ -47,11 +47,13 @@ TRAP_14_HANDLER::
...
     dc.l    .ADD_DEVICE                 ; FC == 10 ; ADD_DEVICE if so...
     dc.l    .DEV_RECVCHAR               ; FC == 11 ; DEV_RECVCHAR if so...
-    dc.l    .DEV_SENDCHAR               ; FC == 12 ; DEV_SENCCHAR if so...
+    dc.l    .DEV_SENDCHAR               ; FC == 12 ; DEV_SENDCHAR if so...
     dc.l    .DEV_CHECKCHAR              ; FC == 13 ; DEV_CHECKCHAR if so...
     dc.l    .EPILOGUE                   ; FC == 14 ; RESERVED
     dc.l    .EPILOGUE                   ; FC == 15 ; RESERVED
     dc.l    .DEV_CTRL                   ; FC == 16 ; DEV_CTRL if so...
+    dc.l    .INPUTCHAR                  ; FC == 17 ; INPUTCHAR if so...
+    dc.l    .CHECKINPUT                 ; FC == 18 ; CHECKINPUT if so...
```
[the remark this answers is not in the document]
Comment (roscopeco) [reply code/firmware/rosco_m68k_firmware/trap14.asm:56]: Good shout, done 👍

File: code/firmware/rosco_m68k_firmware/keyboard.c [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+    chr = try_get_char(device);
+    if (chr != 'b') {
+        return false;
+    }
+
+    chr = try_get_char(device);
+    if (chr != 'd') {
+        return false;
+    }
```
[the remark this answers is not in the document]
Comment (roscopeco) [reply code/firmware/rosco_m68k_firmware/keyboard.c]: Haha fair, I meant to come back and clean this up and forgot - good spot!

**Review by 0xTJ — APPROVED** [review]
I can't yet check the functionality, but looking through the code, I don't see any issues! Looks good!

**Comment by roscopeco** [comment]
> I can't yet check the functionality, but looking through the code, I don't see any issues! Looks good!

Thanks @0xTJ 😄

**Pull request merged**